server/src/main/java/org/opensearch/index/remote/RemoteRefreshSegmentTracker.java

```diff
@@ -422,4 +435,140 @@ void updateUploadTimeMsMovingAverageWindowSize(int updatedSize) {
             this.uploadTimeMsMovingAverageReference.set(this.uploadTimeMsMovingAverageReference.get().copyWithSize(updatedSize));
         }
     }
+
+    public RemoteRefreshSegmentTracker.Stats stats() {
+        return new RemoteRefreshSegmentTracker.Stats(
+            shardId,
+            localRefreshSeqNo,
+            localRefreshTimeMs,
+            remoteRefreshSeqNo,
+            remoteRefreshTimeMs,
+            uploadBytesStarted,
+            uploadBytesSucceeded,
+            uploadBytesFailed,
+            totalUploadsStarted,
+            totalUploadsSucceeded,
+            totalUploadsFailed,
+            rejectionCount.get(),
+            failures.length(),
+            lastSuccessfulRemoteRefreshBytes,
+            uploadBytesMovingAverageReference.get().getAverage(),
+            uploadBytesPerSecMovingAverageReference.get().getAverage(),
+            uploadTimeMsMovingAverageReference.get().getAverage(),
+            getBytesLag()
+        );
+    }
+
+    /**
+     * Represents the tracker's state as seen in the stats API.
+     */
+    public static class Stats implements Writeable {
+
+        public final ShardId shardId;
+        public final long localRefreshCount;
+        public final long localRefreshTimeMs;
+        public final long remoteRefreshCount;
+        public final long remoteRefreshTimeMs;
+        public final long uploadBytesStarted;
+        public final long uploadBytesFailed;
+        public final long uploadBytesSucceeded;
+        public final long totalUploadsStarted;
+        public final long totalUploadsFailed;
+        public final long totalUploadsSucceeded;
+        public final long rejectionCount;
+        public final long consecutiveFailuresCount;
+        public final long lastSuccessfulRemoteRefreshBytes;
+        public final double uploadBytesMovingAverage;
+        public final double uploadBytesPerSecMovingAverage;
+        public final double uploadTimeMovingAverage;
+        public final long bytesLag;
+
+        public Stats(
```

Comment:
should we restrict public access to constructor given the class is only accessed from within the tracker class?

Reply:
done